/*
 * This file was automatically generated by EvoSuite
 * Thu Feb 02 02:30:04 GMT 2023
 */

package org.apache.commons.jxpath.functions;

import org.junit.Test;
import static org.junit.Assert.*;
import static org.evosuite.runtime.EvoAssertions.*;
import java.lang.reflect.Method;
import org.apache.commons.jxpath.JXPathContext;
import org.apache.commons.jxpath.functions.MethodFunction;
import org.apache.commons.jxpath.ri.JXPathContextReferenceImpl;
import org.apache.commons.jxpath.ri.QName;
import org.apache.commons.jxpath.ri.axes.InitialContext;
import org.apache.commons.jxpath.ri.axes.ParentContext;
import org.apache.commons.jxpath.ri.compiler.NodeNameTest;
import org.evosuite.runtime.EvoRunner;
import org.evosuite.runtime.EvoRunnerParameters;
import org.junit.runner.RunWith;

@RunWith(EvoRunner.class) @EvoRunnerParameters(mockJVMNonDeterminism = true, useVFS = true, useVNET = true, resetStaticState = true, separateClassLoader = true, useJEE = true) 
public class MethodFunction_ESTest extends MethodFunction_ESTest_scaffolding {

  @Test(timeout = 4000)
  public void test0()  throws Throwable  {
      MethodFunction methodFunction0 = new MethodFunction((Method) null);
      JXPathContextReferenceImpl jXPathContextReferenceImpl0 = (JXPathContextReferenceImpl)JXPathContext.newContext((Object) methodFunction0);
      InitialContext initialContext0 = (InitialContext)jXPathContextReferenceImpl0.getAbsoluteRootContext();
      QName qName0 = new QName("");
      NodeNameTest nodeNameTest0 = new NodeNameTest(qName0, ".<KB");
      ParentContext parentContext0 = new ParentContext(initialContext0, nodeNameTest0);
      Object[] objectArray0 = new Object[7];
      // Undeclared exception!
      try { 
        methodFunction0.invoke(parentContext0, objectArray0);
        fail("Expecting exception: RuntimeException");
      
      } catch(RuntimeException e) {
         //
         // Cannot invoke null; java.lang.NullPointerException
         //
         verifyException("org.apache.commons.jxpath.functions.MethodFunction", e);
      }
  }

  @Test(timeout = 4000)
  public void test1()  throws Throwable  {
      MethodFunction methodFunction0 = new MethodFunction((Method) null);
      // Undeclared exception!
      try { 
        methodFunction0.toString();
        fail("Expecting exception: NullPointerException");
      
      } catch(NullPointerException e) {
         //
         // no message in exception (getMessage() returned null)
         //
         verifyException("org.apache.commons.jxpath.functions.MethodFunction", e);
      }
  }
}
